src/libraries/System.Private.CoreLib/src/System/HashCode.cs

```diff
@@ -303,7 +303,8 @@ public void Add<T>(T value)
 
         public void Add<T>(T value, IEqualityComparer<T>? comparer)
         {
-            Add(comparer != null ? comparer.GetHashCode(value) : (value?.GetHashCode() ?? 0));
+            // TODO2
+            Add(comparer != null ? comparer.GetHashCode(value!) : (value?.GetHashCode() ?? 0));
```

Comment:
As with the earlier examples with ImmutableHashSet, we should probably fix this implementation.  e.g. today if you wrote code like:
```C#
#nullable enable
using System;

class Program
{
    static void Main(string[] args)
    {
        HashCode c = default;
        c.Add(null, StringComparer.Ordinal);
    }
}
```
it will blow up.  Seems like this line should instead be:
```C#
Add(comparer != null ? comparer.GetHashCode(value ?? 0) : (value?.GetHashCode() ?? 0));
```